### PR TITLE
Feature/code quality measurement tools

### PR DESCRIPTION
**Pull request by MGabor11**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -31,3 +43,25 @@ allprojects {
...
+detekt {
+    debug = true
+    toolVersion = "$detekt_version"
+    input = files("app/src/main/java")
+    filters = ".*/resources/.*,.*/build/.*"
+    reports {
+        xml.enabled = true
+        html {
+            enabled = true
+            destination = file("detekt/detekt.html")
```
Comment (nyekilajos) [inline build.gradle]: It would be much better to generate the output into the build folder. For example ``build/reports/detekt``.

**Comment by nyekilajos** [comment]
As a general comment, it would be nice to solve the problems that detekt reports. Please do that.

File: build.gradle [hunk 2]
```diff
@@ -31,3 +43,25 @@ allprojects {
 task clean(type: Delete) {
     delete rootProject.buildDir
 }
+
+detekt {
```
Comment (nyekilajos) [inline build.gradle]: It would be nice to generate a config file as well, and add to the configuration here.

File: build.gradle (hunk 1 above)
Comment (MGabor11) [reply build.gradle]: Ok. I reset the patch of the report.

File: build.gradle (hunk 2 above)
Comment (MGabor11) [reply build.gradle]: I oganized detekt, and sonarqube task into gradle files.

File: app/detekttask.gradle [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    }
+}
+
+dependencies {
+    detekt "io.gitlab.arturbosch.detekt:detekt-formatting:$detekt_version"
+}
+
+afterEvaluate {
+    tasks.findByName('check')?.dependsOn 'detekt'
+}
```
Comment (nyekilajos) [inline app/detekttask.gradle]: Please use an empty line at the end of the file.

File: app/sonarqubetask.gradle [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        property "sonar.java.coveragePlugin", "jacoco"
+        property "sonar.junit.reportsPath", "build/test-results/debug"
+        property "sonar.android.lint.report", "build/outputs/lint-results"
+        property "sonar.test.inclusions","**/*Test*/**"
+        property "sonar.exclusions","**/*Test*/**"
+        property "sonar.projectKey","weatherapp"
+        property "sonar.host.url","http://localhost:9000"
+    }
+
+}
```
Comment (nyekilajos) [inline app/sonarqubetask.gradle]: Please use an empty line at the end of the file.

File: gradle.properties [hunk 5]
```diff
@@ -13,3 +13,5 @@ org.gradle.jvmargs=-Xmx1536m
 # org.gradle.parallel=true
 android.enableJetifier=true
 android.useAndroidX=true
+
+sonar.login=""
```
Comment (nyekilajos) [inline gradle.properties]: Please use an empty line at the end of the file.

**Pull request merged**